x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/PoolingSessionFactory.java

```diff
@@ -79,9 +80,8 @@
                         + RealmSettings.getFullSettingKey(config, SECURE_BIND_PASSWORD)
                         + "]"
                 );
-            } else {
-                bindPassword = CharArrays.toUtf8Bytes(config.getSetting(LEGACY_BIND_PASSWORD).getChars());
             }
+            bindPassword = CharArrays.toUtf8Bytes(config.getSetting(LEGACY_BIND_PASSWORD).getChars());
```

Comment:
Small readability tweak: moved this out of the `else` block because the corresponding `if` block always throws. With this tweak we avoid a level of nesting